lib/devicetrust/native/device_linux.go

```diff
@@ -51,34 +51,64 @@ var linuxDevice = &tpmDevice{
 }
 
 func enrollDeviceInit() (*devicepb.EnrollDeviceInit, error) {
-	return linuxDevice.enrollDeviceInit()
+	init, err := linuxDevice.enrollDeviceInit()
+	return init, rewriteTPMPermissionError(err)
 }
 
 func signChallenge(chal []byte) (sig []byte, err error) {
-	return linuxDevice.signChallenge(chal)
+	return nil, errors.New("signChallenge not implemented for TPM devices")
 }
 
 func getDeviceCredential() (*devicepb.DeviceCredential, error) {
-	return linuxDevice.getDeviceCredential()
+	cred, err := linuxDevice.getDeviceCredential()
+	return cred, rewriteTPMPermissionError(err)
 }
 
 func solveTPMEnrollChallenge(
 	chal *devicepb.TPMEnrollChallenge,
 	debug bool,
 ) (*devicepb.TPMEnrollChallengeResponse, error) {
+	// No need to call rewriteTPMPermissionError here, enrollDeviceInit must pass
+	// first.
 	return linuxDevice.solveTPMEnrollChallenge(chal, debug)
 }
 
 func solveTPMAuthnDeviceChallenge(
 	chal *devicepb.TPMAuthenticateDeviceChallenge,
 ) (*devicepb.TPMAuthenticateDeviceChallengeResponse, error) {
-	return linuxDevice.solveTPMAuthnDeviceChallenge(chal)
+	resp, err := linuxDevice.solveTPMAuthnDeviceChallenge(chal)
+	return resp, rewriteTPMPermissionError(err)
 }
 
 func handleTPMActivateCredential(encryptedCredential, encryptedCredentialSecret string) error {
 	return errors.New("elevated credential activation not implemented for linux")
 }
 
+func rewriteTPMPermissionError(err error) error {
+	// We are looking for an error that looks roughly like this:
+	//
+	// 	err = &fs.PathError{
+	// 		Path: "/dev/tpmrm0",
+	// 		Err: fs.ErrPermission,
+	// 	}
+	if !errors.Is(err, fs.ErrPermission) {
```

Comment:
Won't the errors.As check below make this check redundant?

Reply:
I suppose we could see PathErrors with other underlying reasons, so I was a bit overzealous here.